exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebUserConnection.java

```diff
@@ -106,7 +110,10 @@ public void sendData(RpcOutcomeListener<Ack> listener, QueryWritableBatch result
         // TODO:  Clean:  DRILL-2933:  That load(...) no longer throws
         // SchemaChangeException, so check/clean catch clause below.
         for (int i = 0; i < loader.getSchema().getFieldCount(); ++i) {
-          columns.add(loader.getSchema().getColumn(i).getName());
+
+          MaterializedField col = loader.getSchema().getColumn(i);
+          columns.add(col.getName());
+          metadata.add(col.getType().getMinorType().name());
```

Comment:
Some types can have precision and scale, does this information needed?

Reply:
Hi @arina-ielchiieva ,
The use case I had in mind was integrating Drill with SQLPad and Apache Superset.  In these instances basically, the UI needed to know if a field was numeric, temporal of any sort, or text so that it could render visualizations properly.  

I'm sure there are other use cases out there, but I know that for me at least, this was a major blocker in getting Drill to work with the various BI tools.  The JDBC interface provided this information, but the RESTful interface did not, so I had to resort to hackery. 

So to answer your question, it might be useful for other use cases to provide precision and scale, but for the one I had in mind, that would not be helpful.

Reply:
I see but though it is not your use case, I think we should consider shipping not only String type but as well information about precision and scale for those who might need it.

Reply:
How would you recommend designing that?  I was trying to keep this PR relatively simple, and backwards compatible, but one option might be to make the metadata a little duplicative so something like:

```
"metadata": [{
   "name": "price",
   "type": "FLOAT4"
   "precision":
   "scale"
   },{
     "name": "customer",
     "type": "VARCHAR"
  ...
]
```
Do you know off hand where the precision/scale or any other attributes of the columns can be accessed?

Reply:
1. Duplicating column name does not make sense.
2. You may not output precision and scale is they are absent, depends in which Object you plan to deserialize this information.
3. Look at major type, for example.